pkg/envoy/ads/secrets.go

```diff
@@ -67,7 +67,7 @@ func makeRequestForAllSecrets(proxy *envoy.Proxy, meshCatalog catalog.MeshCatalo
 
 	// Create an SDS validation cert corresponding to each upstream service that this proxy can connect to.
 	// Each cert is used to validate the certificate presented by the corresponding upstream service.
-	upstreamServices := meshCatalog.ListAllowedOutboundServicesForIdentity(proxyIdentity)
+	upstreamServices := meshCatalog.ListAllowedOutboundServicesForIdentity(proxyIdentity.ToServiceIdentity())
```

Comment:
we might want to rename `proxyIdentity` to `serviceAccount` to make this more clear.

Reply:
@michelleN I renamed `proxyIdentity` to `serviceAccount`.  Also left a comment - we need to continue work on this as part of:
  - [ ] **certificates: Envoy xDS Certs should have a ServiceIdentity in the CN; not ServiceAccount** #3186